src/modules/dashboard/components/Dialogs/ManageWhitelistDialog/ManageWhitelistDialogForm.tsx

```diff
@@ -181,7 +181,12 @@ const ManageWhitelistDialogForm = ({
         <Button
           appearance={{ theme: 'secondary', size: 'large' }}
           onClick={back}
-          text={{ id: 'button.back' }}
+          text={{
+            id:
+              back.toString() === `() => callStep(prevStep)`
+                ? 'button.back'
+                : 'button.cancel',
```

Comment:
Maybe it will be better to do something like that:
'whateverButtonText' - with default to 'button.back'
and then just put 'button.cancel' for this case

I dont really like this approach of converting to string and comparing if this is that function

Reply:
Haha, yeah, I don't like it either. I will try it as you suggested.